cluster/kube-util.sh

```diff
@@ -16,10 +16,16 @@
 
 # This script contains skeletons of helper functions that each provider hosting
 # Kubernetes must implement to use cluster/kube-*.sh scripts.
-# It sets KUBERNETES_PROVIDER to its default value (gce) if it is unset, and
+# It sets KUBERNETES_PROVIDER to its default value (gce) if it is unset
+# and KUBERNETES_CONFORMANCE_TEST is unset, and
 # then sources cluster/${KUBERNETES_PROVIDER}/util.sh.
 
-KUBERNETES_PROVIDER="${KUBERNETES_PROVIDER:-gce}"
+if [[ -n "${KUBERNETES_CONFORMANCE_TEST:-}" ]]; then
+    KUBERNETES_PROVIDER=""
```

Comment:
I'm not sure why we do that. if anything, I'd be for removing that line in ginkgo-e2e.sh.